New build using webpack (WIP, creating PR to deploy to dev for testing)

cheapsteak opened the pull request:
regressions:
- [x]  ![image](https://cloud.githubusercontent.com/assets/743976/19040901/023aba66-8954-11e6-97cb-cfea18a17f45.png)

cheapsteak committed:
webpack is building and running. page is broken. vendor files need work

cheapsteak committed:
more fixing weird js errors

cheapsteak committed:
bamiobio, vcfiobio, releases page

cheapsteak committed:
revert to untouched genome viewer

cheapsteak committed:
shim libraries

cheapsteak committed:
install ng-annotate-loader

cheapsteak committed:
remove script! loaders from index.js

cheapsteak committed:
remove iffe from app.js

cheapsteak committed:
build now works with styles

cheapsteak committed:
revert: add back variables.scss

cheapsteak committed:
move css imports out of index.html and into styles.scss

cheapsteak committed:
copy files that are expected to be in the build dest folder

cheapsteak committed:
add scripts and styles for 404 page

dcc-jenkins commented:
Deployed to https://dev.dcc.icgc.org/portals/134 for testing

btiernay commented:
Please update maven pom to adjust the build steps.  Also please add some info to the repo markdown file for the module. Thanks

btiernay commented on dcc-portal-ui/.jshintrc:
```diff
@@ -15,7 +15,7 @@
   "undef": true,
   "unused": true,
-  "strict": true,
+  "strict": false,
```
Intentional?

cheapsteak replied:
Everything is implicitly strict now with the bundler, no need to write it ourselves, can't even disable it

btiernay commented on dcc-portal-ui/app/scripts/common/styles/_header.scss:
```diff
@@ -432,13 +431,13 @@ header {
-            weight: 300;
-            family: $sans-stack;
-          } */
+        // /*@include breakpoint($break-tablet) {
```
Interesting

cheapsteak replied:
going to remove these comments before PR is sent for merging, just wanted to see what the jar would build to

/**/ apparently sometimes break the build with node-sass

btiernay commented on dcc-portal-ui/app/scripts/vendor.js:
```diff
@@ -0,0 +1,84 @@
+require('expose?ZeroClipboard!zeroclipboard/dist/ZeroClipboard.js');
+require('angular-gettext/dist/angular-gettext.min.js');
+
+
```
Why so generous with this extra line?

btiernay commented on dcc-portal-ui/app/vendor/styles/bamiobio/flat-ui.scss:
```diff
@@ -1,8 +1,8 @@
   font-family: 'Flat-UI-Icons';
-  src: url('../fonts/Flat-UI-Icons.eot');
-  src: url('../fonts/Flat-UI-Icons.eot?#iefix') format('embedded-opentype'), url('../vendor/styles/bamiobio/fonts/Flat-UI-Icons.woff') format('woff'), url('../vendor/bamiobio/fonts/Flat-UI-Icons.ttf') format('truetype'), url('../vendor/bamiobio/fonts/Flat-UI-Icons.svg#Flat-UI-Icons') format('svg');
+  // src: url('../vendor/styles/bamiobio/fonts/Flat-UI-Icons.eot');
```
If this is not needed, should we remove?

btiernay commented on dcc-portal-ui/config/shims/genome-viewer/exports.js:
```diff
@@ -0,0 +1,134 @@
+/*
+
+To generate this array, 
```
Nice comment. Sad so much wrestling required.

btiernay commented on dcc-portal-ui/app/styles/_base/_base.scss:
```diff
@@ -19,12 +19,22 @@
+
+@import "~breakpoint-sass";
+@import "~singularitygs";
+// we don't seem to be using singularity-extras anywhere
```
We like to capitalize our comments in our family, son.

btiernay commented on dcc-portal-ui/tasks/utils/chrome.applescript:
```diff
@@ -0,0 +1,39 @@
+on run argv
```
Sup with this file?

cheapsteak replied:
When the local dev tasks runs, this opens a new window if one hasn't been opened previously, or focuses on the previously opened tab (instead of opening a new tab)

btiernay commented on dcc-portal-ui/config/paths.js:
```diff
@@ -0,0 +1,20 @@
+var path = require('path');
+
+
+// function resolveOwn(relativePath) {
```
Commented out code makes baby Jesus cry.

btiernay commented on dcc-portal-ui/app/scripts/app/js/app.js:
```diff
@@ -684,5 +678,3 @@
   function gettext(string){
     return string;
   }
-
```
Nice :)

btiernay commented on dcc-portal-ui/app/styles/_base/_normalize.scss:
```diff
@@ -1,628 +0,0 @@
-/* normalize.css v2.1.2 | MIT License | git.io/normalize */
```
Where did this go?

cheapsteak replied:
using a library now (check base.scss （or styles.scss))

btiernay replied:
Could it have changed the look and feel?

cheapsteak replied:
not noticeably

btiernay commented on dcc-portal-ui/app/vendor/scripts/bamiobio/binary.js:
```diff
@@ -1,1332 +0,0 @@
-/*
```
And this?

cheapsteak replied:
exteralized using a library instead

btiernay replied:
Any perceived differences in the site's appearance?

btiernay commented on dcc-portal-ui/app/vendor/scripts/bamiobio/class.js:
```diff
@@ -1,108 +0,0 @@
-/*
```
etc

cheapsteak replied:
externalized to a library

cheapsteak commented:
Sorry should have mentioned it wasn't ready for code review yet (didn't expect you to be looking at this afterhours ._.), just wanted to see what it'd look like built and run as a jar (without having to stick around and wait for it to actually build)

Should we maybe add "branches" as "candidates" for the dev server so the raison d'etre of PRs can remain pure?

cheapsteak committed:
minor: remove commented out code

cheapsteak committed:
remove extra linebreaks

cheapsteak committed:
minor: remove commented out code

cheapsteak committed:
remove unused readVendors task

cheapsteak committed:
update pom.xml with correct build command

cheapsteak committed:
Remove react-related eslint rules

cheapsteak committed:
Add linting

cheapsteak committed:
update output path

cheapsteak committed:
explicitly set $ICGC_DEV_CONFIG to global

cheapsteak committed:
minor:eslint: add explicit return to reduce

cheapsteak committed:
minor:eslint:whitespace and linebreaks

cheapsteak committed:
Revert back to including icgc-icons.css in index.html

cheapsteak committed:
enable eslint loader

cheapsteak committed:
Switch to normalize.css@3.0.1, looks closer to current site

cheapsteak committed:
copy favicon

cheapsteak committed:
remove jshint

Commit:
Updated pom to run UI npm tasks on compile.

cheapsteak committed:
bugfix:deps: add missing deps

cheapsteak committed:
run bower install in npm’s postinstall

cheapsteak committed:
remove grunt deps made obsolete by webpack buildstep

cheapsteak committed:
update node dependency version

cheapsteak committed:
add icgc-icons.css to list of things to copy on build

cheapsteak committed:
Use ExtractTextPlugin for scss files

cheapsteak committed:
add CommonsChunkPlugin to prod webpack config

cheapsteak committed:
add karma task to package.json

cheapsteak committed:
deps: add flag-icon-css

cheapsteak committed:
use flag-icons from node_modules

cheapsteak committed:
remove flag-icon files

cheapsteak committed:
deps: remove unused grunt deps

cheapsteak committed:
deps: add karma plugins for babel and webpack

dcc-jenkins commented:
Deployed to https://dev.dcc.icgc.org/portals/149 for testing

cheapsteak committed:
use normalize.css v2.1.2, switch to bower for dep

cheapsteak committed:
move pql unit test out of unit/directory since they aren’t being used

cheapsteak committed:
update karma.conf.js to work with webpack

cheapsteak committed:
remove grunt-karma

cheapsteak committed:
remove unused karma e2e config

cheapsteak committed:
rename bower script to “bower” (no longer postinstall)

cheapsteak committed:
add npm run bower to pom

cheapsteak committed:
add normalize-css to bower.json

cheapsteak committed:
remove highcharts from bower. upgrade to 5. fixes DCC-5315

cheapsteak committed:
export saveAs from FileSaver.js. fixes DCC-5318

cheapsteak committed:
add vendor styles to copy path

cheapsteak committed:
deps: use font-awesome deps from npm

cheapsteak committed:
css: bring content div up to cover up aside. fixes DCC-5317

cheapsteak committed:
bugfix: init local global `key` for gv-config. fixes DCC-5324

cheapsteak committed:
<article>s next to .t_sidebar will be pos:relative

sometimes they’re not directly under .content, but wrapped in another div (e.g. in /browser)

btiernay commented:
WooT!
